lib/monitor.toit

```diff
@@ -246,13 +246,15 @@ monitor ResourceState_:
       group_ = null
       remove_finalizer this
 
-  // Called when the state changes because of the call to
-  // [register_object_notifier] in the constructor.
+  // Called on timeouts and when the state changes because of the call
+  // to [register_object_notifier] in the constructor.
   notify_:
-    if not resource_: return
-    state := read_state_ group_ resource_
-    state_ |= state
-    // TODO(kasper): Only notify if we added new bits?
+    resource := resource_
+    if resource:
+      state := read_state_ group_ resource
+      state_ |= state
+    // Always call the super implementation to avoid getting
```

Comment:
This may not have been a real problem, because leaving the dispose call will notify waiters (state changed) and so anyone waiting for resource changes will be unblocked and the condition rechecked. However, it feels iffy to not let the super class know about the notification given that it is the super class that requested the timeout/cancelation notifications in the first place.